language/stackless-bytecode/tree_heap/output.bpl

```diff
@@ -251,6 +264,26 @@ procedure {:inline 1} Not(src: Value) returns (dst: Value)
     dst := Boolean(!b#Boolean(src));
 }
 
+procedure {:inline 1} Eq(src1: Value, src2: Value) returns (dst: Value)
+{
+    dst := Boolean(src1 == src2);
+}
+
+procedure {:inline 1} Neq(src1: Value, src2: Value) returns (dst: Value)
+{
+    dst := Boolean(src1 != src2);
+}
+
+procedure {:inline 1} Eq_Vector_T(src1: Value, src2: Value) returns (dst: Value)
+{
+    dst := Boolean(src1 == src2);
+}
```

Comment:
Why doesn't this need to check equality only for the elements with indices 0...l-1? 
I'm worried about a case where:
v = empty vector
push(&mut v, 49)
pop(&mut v)
assert(v == \old(v))
The full vectors will differ at index 0, since \old(v) has the original value and v[0] == 49.

Reply:
Right it is a problem. This version is only a placeholder. To check all elements from 0 to l-1, we have to use loops or recursion right? I'm not sure how well that's gonna work out. 

Maybe we can initialize values in all the cells to some default value when we create an empty vector(using lambda?). And when we pop something off a vector, we change the value at that index to the same default value?